semgrep/semgrep/core_runner.py

```diff
@@ -224,7 +228,18 @@ def _run_rule(
         for language, all_patterns_for_language in self._group_patterns_by_language(
             [rule]
         ).items():
-            targets = target_manager.get_files(language, rule.includes, rule.excludes)
+            try:
+                targets = target_manager.get_files(
+                    language, rule.includes, rule.excludes
+                )
+            except UnknownLanguageError as ex:
+                raise ErrorWithSpan(
```

Comment:
It might be easier to know what errors are raised by semgrep if we subclass and have an InvalidLanguageError that takes in language and span?

Reply:
We can do that but it would duplicate what you have -- at the time you throw the error we don't have context.

Since the cause is chained in, the verbose mode I'm imagining would be able to easily show a complete stack trace.

However, we can definitely just subclass and have two unknown language errors.

Reply:
Oh I see. I think that's fine. Maybe rename the target manager exception to _UnknownLanguageError or something since with this PR that exception should be internal anyway. I think it makes sense as you say below to always subclass ErrorWithSpan so it's always clear from the type of the exception alone what kind of error this is vs having to look into the short_msg to find out